lib/sax.js

```diff
@@ -576,18 +576,19 @@ function parseDCC(source, start, domBuilder, errorHandler) {
 }
 
 function parseInstruction(source, start, domBuilder) {
-	var end = source.indexOf('?>', start);
-	if (end) {
-		var match = source.substring(start, end).match(/^<\?(\S*)\s*([\s\S]*?)\s*$/);
-		if (match) {
-			domBuilder.processingInstruction(match[1], match[2]);
-			return end + 2;
-		} else {
-			//error
-			return -1;
+	var match = source.substring(start).match(grammar.PI_match);
+	if (!match) {
+		throw new ParseError('Invalid processing instruction starting at position ' + start);
```

Comment:
the errors thrown in this method should also be using the `errorHandler.fatalError` instead of throwing themselves.

Reply:
Done. I thought ParseError was expected here since all the code is in a try catch block which catches and rethrows ParseError.